testing/xctest.md

```diff
@@ -0,0 +1,36 @@
+# XCTest
+
+1. [Unit Test Strategy](#Unit-Test-Strategy))
+1. [Using `final` for `XCTestCase`](#Using-final-for-XCTestCase) 
+1. [setUp, tearDown and member variables](#setUp-tearDown-and-member-variables)
+1. [Dummys, Fakes, Stubs, Spies and Mocks](#Dummys-Fakes-Stubs-Spies-and-Mocks)
+
+## Unit Test Strategy
+When working on a codebase with a team of any size, the modern convention is to unit test as much as possible. If you're in any doubt about whether you should write a unit test ask yourself: if my code changes were to be accidentally deleted by a merge conflict resolution or such like, would the test suits fail? If you're making a change to UI Layout then snapshot tests can act as a red flag if someone is to change your layout by accident later on.
+
+It is rare that something cannot or should not be unit tested, but if you stumble across one of those situations, try to ensure that if your code is removed or altered erroneously, either the compiler will refuse to compile or there is a UI/Integration tests somewhere that will fail.  Test coverage tools can usually identify un-tested code but try not to lean on them, good unit testing leads to well structured SOLID code.
+
+## Using `final` for `XCTestCase`
+Adding the `final` keyword to your class is an indicator to the reader that this class is not meant to be extended. This makes a lot of sense for `XCTestCase`. There are only a few situations where you would actually want to extend your `XCTestCase` class, but usually if you are doing this then it is probably a code smell that you're doing something wrong.
+
+`final` also has performance implications. This means you want to try and make as many of your classes `final` as possible.  Google will tell you more about this. __Note: it's possible to achieve the same performance gains by making all member variables private__
```

Comment:
I'd be keen we lead people to authoritative sources directly, also helping document why we believe something at the time of writing.
```suggestion
`final` also has [performance implications](https://developer.apple.com/swift/blog/?id=27). This means you want to try and make as many of your classes `final` as possible. __Note: it's possible to achieve the same performance gains by making all member variables private__
```

I think the pattern benefits are enough justification for this policy, to be honest.